jib-core/src/main/java/com/google/cloud/tools/jib/frontend/BuildStepsRunner.java

```diff
@@ -255,6 +257,14 @@ public void build(HelpfulSuggestions helpfulSuggestions) throws BuildStepsExecut
         throw new BuildStepsExecutionException(
             helpfulSuggestions.forInsecureRegistry(), exceptionDuringBuildSteps);
 
+      } else if (exceptionDuringBuildSteps instanceof RegistryErrorException) {
+        // RegistryErrorExceptions have good messages
+        RegistryErrorException registryErrorException =
```

Comment:
Is this cast actually necessary? I think you should be able to get the same message without the cast.

Reply:
Yeah it's necessary as `exceptionDuringBuildSteps` is declared as `Throwable`.

Reply:
Oh gotcha